tests/pytorch/test_stream.py

```diff
@@ -0,0 +1,39 @@
+from dgl import rand_graph
+import dgl._ffi.streams as FS
+import dgl.ops as OPS
+import unittest
+import backend as F
+import torch
+
+
+@unittest.skipIf(F._default_context_str == 'cpu', reason="stream only runs on GPU.")
+def test_basics():
+    g = rand_graph(10, 20, device=F.cpu())
+    x = torch.ones(g.num_nodes(), 10)
+
+    # launch on default stream fetched via torch.cuda
+    s = torch.cuda.default_stream(device=F.ctx())
+    with torch.cuda.stream(s):
+        xx = x.to(device=F.ctx(), non_blocking=True)
+    with FS.stream(s):
+        gg = g.to(device=F.ctx())
+    s.synchronize()
+    OPS.copy_u_sum(gg, xx)
+
+    # launch on new stream created via torch.cuda
+    s = torch.cuda.Stream(device=F.ctx())
+    with torch.cuda.stream(s):
+        xx = x.to(device=F.ctx(), non_blocking=True)
+    with FS.stream(s):
+        gg = g.to(device=F.ctx())
+    s.synchronize()
```

Comment:
Is it possible to check if the stream is indeed non-default in unit tests?

Reply:
If you mean the created stream `s`, s.cuda_stream is nullptr if default, otherwise non-nullptr.
If you mean g.to() is really running on the stream we set, the most reliable way is printing something when doing GPUCopy which is not easy or elegant to implement. the following OPS.copy_u_sum() I called is an indirect way to check g.to() running on the stream we set, though it's not very reliable I think.

which one do u mean here?